implementations/rust/ockam/ockam_node/src/storage/database/sqlx_from_row_types.rs

```diff
@@ -0,0 +1,80 @@
+use sqlx::database::HasValueRef;
+use sqlx::error::BoxDynError;
+use sqlx::postgres::any::AnyTypeInfoKind;
+use sqlx::{Any, Database, Decode, Type, Value, ValueRef};
+
+/// This type is used to map Option<T> fields for the types deriving `FromRow`
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct Nullable<T>(Option<T>);
```

Comment:
Should we also implement `From<Option<T>> for Nullable<T>` and `From<Nullable<T>> for Option<T>`?

Reply:
This type is only used in one direction `Nullable<T>` to `Option<T>`. So we could implement `From<Nullable<T>> for Option<T>`. The problem is that this changes `to_option()` to `.clone().into()` in many places, see my next comment.